### PR TITLE
common: Output logs to stderr when no journal

### PR DESCRIPTION
**Pull request by petervo**

### CODE REVIEW

**Comment by stefwalter** [comment]
So to summarize, the SELinux issues here are a bummer.

**Comment by petervo** [comment]
@stefwalter looks like selinux lets us check if the socket file exists. WDYT?

File: src/common/cockpitlog.c [hunk 1]
```diff
@@ -135,6 +155,13 @@ cockpit_set_journal_logging (const gchar *stderr_domain,
...
+  /* SELinux won't let us always open the sd_journal_stream_fd
+   * so just check that the main journal socket exists
+   */
+  have_journal = g_file_test ("/run/systemd/journal/socket", G_FILE_TEST_EXISTS);
+  if (only)
+    old_handler = NULL;
+
   if (only && stderr_domain)
     {
       fd = sd_journal_stream_fd (stderr_domain, LOG_WARNING, 0);
```
Comment (stefwalter) [inline src/common/cockpitlog.c]: What does this return when no journal is available?

Comment (petervo) [inline src/common/cockpitlog.c]: That's what the original version of this PR did, but ```sd_journal_stream_fd``` in the cockpit-ws context caused selinux errors. We know we are in the cockpit-ws context when stderr_domain is NULL. That's why this version works and the original didn't.

**Pull request closed without merging**